### PR TITLE
feat: refactor into packages and publish skeleton analyzer plugin FUI-1512

### PR DESCRIPTION
**Pull request by matteematt**
📓  &nbsp; **Related Issue**
If this PR is related to an issue then link it here, or delete this section.

🤔  &nbsp; **What does this PR do?**

Restructure the project as a monorepo and use lerna to publish multiple packages
- Restructure the project using `npm workspaces` 
- Enable lerna for publishing

📑  &nbsp; **How should this be tested?**

Update test steps to match your PR:

```
1. Checkout branch
2. `npm run bootstrap`
3. `npm run test:unit`

To test the LSP is working correctly in vscode
1. `cd packages/showcase/example`
2. `code .` 
3. Open a typescript file such as `root.ts`
4. In the Command Palette choose `TypeScript: select typescript version...` and then choose the workspace version
5. The LSP should be enabled. You will need to make a code change after the LSP is enabled before you see any Intellisense.
```

> These testing instructions assume that you've already setup the LSP in your IDE with the `example` app. If you haven't then follow the instructions in the `README.md`.

✅  &nbsp; **Checklist**

- [X] I have added tests for my changes.
- [X] I have updated the project documentation.
- [X] I have followed the [contribution guidelines](https://github.com/genesiscommunitysuccess/custom-elements-lsp/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -3,21 +3,23 @@ module.exports = {
   plugins: ['@typescript-eslint', 'eslint-plugin-import', 'unused-imports'],
   parserOptions: {
     project: [
-      './tsconfig.json',
-      './example/src/tsconfig.json',
-      './example-lib/tsconfig.json',
-      './src/analyzer-import-alias-plugin/tsconfig.json',
+			'./packages/core/custom-elements-lsp/tsconfig.json',
```
Comment (MrBrunoWolff) [inline .eslintrc.js:6]: nit: "interesting" indentation here!

File: .eslintrc.js [hunk 2]
```diff
@@ -3,21 +3,23 @@ module.exports = {
...
   },
   extends: ['google', 'plugin:prettier/recommended', 'plugin:import/typescript'],
   ignorePatterns: [
     'custom-elements-manifest.config.js',
+    'commitlint.config.js',
     '.eslintrc.js',
     '**/dist/**/*',
     '**/public/**/*',
     '**/out/**/*',
+		'scripts/**/*'
```
Comment (MrBrunoWolff) [inline .eslintrc.js:22]: nit: same here

**Review by MrBrunoWolff — COMMENTED** [review]
First pass = just nitpicking, looks good

CI seems to be failing, will wait until that's green.

File: packages/core/custom-elements-lsp/src/index.ts [hunk 3]
```diff
@@ -0,0 +1,3 @@
+import { init } from './plugin';
+
+export = init;
```
Comment (cistov) [inline packages/core/custom-elements-lsp/src/index.ts:3]: In commonJS you'd assign to `module.exports`. In ES6 `export` is a keyword - you don't assign to it. You can do smth like `export { init as default } from './plugin';

https://developer.mozilla.org/en-US/docs/web/javascript/reference/statements/export

File: packages/core/custom-elements-lsp/package.json [hunk 4]
```diff
@@ -0,0 +1,91 @@
+{
+  "name": "@genesiscommunitysuccess/custom-elements-lsp",
+  "version": "2.0.1",
+  "license": "MIT",
+  "main": "./out", "bin": {
```
Comment (cistov) [inline packages/core/custom-elements-lsp/package.json:5]: Funky formatting

File: packages/core/custom-elements-lsp/package.json [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+    "access": "public"
+  },
+  "homepage": "https://github.com/genesiscommunitysuccess/custom-elements-lsp",
+  "bugs": {
+    "url": "https://github.com/genesiscommunitysuccess/custom-elements-lsp/issues"
+  },
+  "scripts": {
+    "bootstrap:module": "npm run build",
+    "build": "npx tsc -b ./src/tsconfig.json",
+    "build:watch": "npx tsc --watch",
```
Comment (cistov) [inline packages/core/custom-elements-lsp/package.json:39]: You don't need `npx` here

File: package.json [hunk 6]
```diff
@@ -35,25 +16,22 @@
...
-    "bootstrap:ci": "npm run bootstrap:module:ci && cd example-lib && npm run bootstrap:module:ci && cd ../example && npm run bootstrap:module:ci && cd .. ",
+    "bootstrap": "npm run bootstrap:husky && npm i && npx lerna run bootstrap:module",
+    "bootstrap:ci": "npm run bootstrap:husky && npm ci && npx lerna run bootstrap:module",
     "bootstrap:husky": "npx --yes husky install",
-    "bootstrap:module": "npm i && npm run bootstrap:husky",
-    "bootstrap:module:ci": "npm ci && npm run bootstrap:husky",
     "build": "npx tsc",
     "build:watch": "npx tsc --watch",
-    "clean": "rm -rf example-lib/node_modules/ && rm -rf example/node_modules/ && rm -rf node_modules/",
+    "clean": "npx lerna run clean:dist && npm run clean:module",
```
Comment (cistov) [inline package.json:24]: No need for `npx` for tsc/lerna

File: packages/core/custom-elements-lsp/package.json [hunk 7]
```diff
@@ -1,91 +1,92 @@
...
-    "eslint-plugin-prettier": "^4.2.1",
-    "eslint-plugin-unused-imports": "^2.0.0",
-    "husky": "^8.0.3",
-    "jest": "^29.5.0",
-    "lint-staged": "^13.2.2",
-    "semantic-release": "21.0.7",
-    "ts-jest": "^29.1.0",
-    "typescript": "^4.4.3"
-  }
+	"name": "@genesiscommunitysuccess/custom-elements-lsp",
```
Comment (MrBrunoWolff) [inline packages/core/custom-elements-lsp/package.json:2]: nit: 

there's something happening with the formatting here 👀

Comment (matteematt) [reply packages/core/custom-elements-lsp/package.json]: Fixed in latest push

**Review by cistov — APPROVED** [review]

File: packages/core/custom-elements-lsp/src/index.ts [hunk 8]
```diff
@@ -0,0 +1,3 @@
+import { init } from './plugin';
+
+module.exports = init;
```
Comment (cistov) [inline packages/core/custom-elements-lsp/src/index.ts:3]: Mixing ES6 and CommonJS in the same module - should use one or the other

Comment (matteematt) [reply packages/core/custom-elements-lsp/src/index.ts:3]: Will get to this another time but going to merge to try out the publishing now. Thanks.

**Pull request merged**